ocamltest/ocaml_toplevels.ml

```diff
@@ -1,70 +0,0 @@
-(**************************************************************************)
-(*                                                                        *)
-(*                                 OCaml                                  *)
-(*                                                                        *)
-(*             Sebastien Hinderer, projet Gallium, INRIA Paris            *)
-(*                                                                        *)
-(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
-(*     en Automatique.                                                    *)
-(*                                                                        *)
-(*   All rights reserved.  This file is distributed under the terms of    *)
-(*   the GNU Lesser General Public License version 2.1, with the          *)
-(*   special exception on linking described in the file LICENSE.          *)
-(*                                                                        *)
-(**************************************************************************)
-
-(* Description of the OCaml toplevels *)
-
-open Ocamltest_stdlib
-
-class toplevel
-  ~(name : string)
-  ~(flags : string)
-  ~(directory : string)
-  ~(exit_status_variable : Variables.t)
-  ~(reference_variable : Variables.t)
-  ~(output_variable : Variables.t)
-  ~(backend : Ocaml_backends.t)
-  ~(compiler : Ocaml_compilers.compiler)
-= object (self) inherit Ocaml_tools.tool
-  ~name:name
-  ~family:"toplevel"
-  ~flags:flags
-  ~directory:directory
-  ~exit_status_variable:exit_status_variable
-  ~reference_variable:reference_variable
-  ~output_variable:output_variable
-  as tool
-  method backend = backend
-  method compiler = compiler
-  method ! reference_file env prefix =
-    let default = tool#reference_file env prefix in
-    if Sys.file_exists default then default else
-    let suffix = self#reference_filename_suffix env in
-    let mk s = (Filename.make_filename prefix s) ^ suffix in
-    let filename = mk
-      (Ocaml_backends.string_of_backend self#backend) in
```

Comment:
This is slightly different from the `ocaml_compilers.ml` override, in that here the backend is used instead of the target.

Reply:
In fact I believe that the difference comes from the way `tool#reference_file env prefix` (the `default` choice) uses the `directory` variable, which for the toplevel is equal to `"ocaml"`.

Reply:
Indeed! Fix pushed (+ Changes entries).